requirements.txt

```diff
@@ -1,4 +1,4 @@
 sqlite3@3.17 -DCMAKE_POSITION_INDEPENDENT_CODE=On
 boost@1.72 -DCMAKE_POSITION_INDEPENDENT_CODE=On --build
 half,https://github.com/pfultz2/half/archive/1.12.0.tar.gz -X header -H sha256:0a08660b68abb176ebc2a0cdf8de46e3182a7f46c66443bb80dbfaaec98cf969 --build
-ROCmSoftwarePlatform/llvm-project-mlir@release/rocm-5.1 -DBUILD_FAT_LIBMLIRMIOPEN=1
+ROCmSoftwarePlatform/llvm-project-mlir@b04c00925237718e5d260efb2a1776ee6f99f5b4 -DBUILD_FAT_LIBMLIRMIOPEN=1
```

Comment:
@whchung could you check the CMakeLists file in MLIR?
```
CMake Error at external/llvm-project/mlir/lib/ExecutionEngine/CMakeLists.txt:193 (message):

No non-CPU ROCm agents found on the system, and ROCM_TEST_CHIPSET is not
 defined

Reply:
@junliume This seems to be a test node-specific issue, from the error message it seems to denote no GPU is found on the test node.

Reply:
@junliume Wondering what's the ROCm version used in the CI of MIOpen? This particular commit was added today (2/24/2022) to migrate everything on MLIR CI infrastructure to ROCm 5.0 if I understand correctly.

Reply:
@whchung MIOpen CI has not yet upgraded to ROCm 5.0 (we are still using 4.3.1) but will do so soon.
@okakarpa Can we start testing on CI upgrade for a selected few nodes first, and plan on moving all CI nodes to 5.0.1 eventually? Thanks! If a meeting is needed between MIOpen and MIGraphX folks, please feel free to arrange one. Thanks!

Reply:
@whchung The commit is not an issue. That commit will upgrade our CI's ROCm version and has no impact on our source code or how MIOpen builds our static library.

Judging by that error, it points to the machine executor `rocm_agent_enumerator` return an empty agent list. The reason why happened is due to this is an opencl environment and that we shouldn't presume `rocm_agent_enumerator` to must exist in an client environment. The must have been an issue brought by the recent upstream merge and has to be reverted. The reason why we didn't catch it locally is due to we don't test opencl builds in our CI.

Reply:
@jerryyin Thanks. It's done.

Reply:
@jerryyin I'd actually prefer `MLIR_ENABLE_ROCM_RUNNER=0` to be set in MLIR's wildcard build with `BUILD_FAT_LIBMLIRMIOPEN` and regressed in MLIR CI first. If we just change it here it will create discrepancy in how MIOpen use our static library vs how our CI test it. So the right sequence to do this should be:
 1. Submit a new PR in MLIR that make the rocm runner turned off in the top level cmake
 2. Have that pass our CI, got it merged
 3. Update the commit hash in this PR

Reply:
MLIR PR filed: https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/pull/529

Reply:
@jerryyin Per discussion and https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/miopen-dialect/external/llvm-project/mlir/lib/ExecutionEngine/CMakeLists.txt#L185-L199 I think the earlier assessment was incorrect. It's still a node-specific issue, not related with the logic introduced in this PR. I've closed https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/pull/529 .

Reply:
Now https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/pull/531 has been merged, I've moved the pointer in this PR.